### PR TITLE
Allow to set the volume name on creation as the first argument.

### PR DESCRIPTION
**Pull request by calavera**
`docker volume create` requires a flag to set the volume name. This is
specially confusing compared with `docker network create` that uses the
first argument in the command to set the name.

This change allows people to set the volume name with the first
argument, copying what `docker network create` does. It preserves the
`--name` flag and uses it over the argument if it's set. That way we
don't break current workflows.

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

**Comment by duglin** [comment]
If we do this then I recommend that we deprecate the --name flag

**Comment by thaJeztah** [comment]
+1 I actually opened an issue for that a while back https://github.com/docker/docker/issues/17194. I closed that one because "name" being optional was the reason for not doing this in the first place, but seeing this, I think it makes sense.

+1 to marking `--name` as deprecated as well

**Comment by thaJeztah** [comment]
@duglin can I take your comment as a "+1" on design?

/cc @cpuguy83

**Comment by duglin** [comment]
@thaJeztah it depends on whether "name" is now a required property of a volume. If so, then SGTM. If not then my initial reaction is to keep optional info behind flags - like we do for `docker run --name ...`

**Comment by thaJeztah** [comment]
Hm, not sure. I think the main reason for using `--name` on `docker run` is that making the name optional in that command `docker run --options [name] image` would be more confusing, and possibly hard to parse? For `volume create` its not that confusing.

**Comment by duglin** [comment]
Until another optional or required arg comes up - then we're in trouble. Keeping optional things behind flags allows for more future freedom and less ambiguity.

**Comment by vdemeester** [comment]
I agree with @duglin on that one. If the name is now required for a volume, then it makes sence, but if it's optional (which is the case now), this feels weird…

But I do agrees with @calavera that it feels strange to have a flag `--name` for `volume` subcommand and not for `network` (have a lot of question about that).

**Comment by calavera** [comment]
I'm going to close this one. I still feel like it's weird, but I'm not completely sure it's worth changing. Thanks for your comments, great feedback! :raised_hands: :pray:
